web/modules/custom/joinup_communities/joinup_collection/src/Plugin/pipeline/Pipeline/JoinupCollectionPipeline.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Drupal\joinup_collection\Plugin\pipeline\Pipeline;
+
+use Drupal\joinup_federation\JoinupFederationPipelinePluginBase;
+
+/**
+ * The data pipeline for the Joinup collection.
+ *
+ * @PipelinePipeline(
+ *   id = "joinup_collection",
+ *   label = @Translation("Joinup collection - ADMSv1 > ADMSv2"),
+ *   steps = {
+ *     "manual_upload",
+ *     "remove_unsupported_data",
+ *     "add_joinup_vocabularies",
+ *     "adms_validation",
+ *     "user_selection_filter" = {
+ *       "collection" = "http://placeHolder/11c81d8f-1527-4044-a694-b847d66362e1",
```

Comment:
I was thinking about it as well. Indeed having the same id as configuration in more than one steps is not that useful. We should indeed pass this to the pipeline configuration. However, the way I was thinking it is that in the future, we could have a configuration screen for the pipeline.
For example, we could have a single configuration screen where the user selects the collection and has the ability to choose extra steps to run. The steps' weight is fixed (because we want each step to run at a specific point always) but the user has the option to let's say add a conversion step, or add a custom cleanup step that we have created.